xtask/src/commands/prep/schema.rs

```diff
@@ -0,0 +1,129 @@
+use std::fs;
+
+use anyhow::{anyhow, Result};
+use camino::Utf8PathBuf;
+use reqwest::blocking::Client;
+use rover_std::Fs;
+use uuid::Uuid;
+
+const SCHEMA_DIR: &str = "./crates/rover-client/.schema";
+
+/// This script downloads the schema if it's not in the file system
+/// or if we can detect the user is internet connected.
+///
+/// If the user is offline and the schema already exists in the file system, the script does nothing.
+///
+/// The URL to fetch the schema can be overridden with the APOLLO_GPAPHQL_SCHEMA_URL environment variable.
+///
+/// Note: crate::info! statements only show up with `cargo build -vv`
```

Comment:
Is `xtask prep` called from within `cargo build`?

Reply:
nope, but we could get it to if we wanted to. right now i have it set up so that if the schema file doesn't exist at all we'll get an error telling us to run that command. maybe we should just run it directly.

Reply:
So do the `crate::info!` calls do anything? Are those just proxies to `log` or `tracing`?

Reply:
oh yeah this comment is outdated - good catch